docs/_posts/2015-08-26-architecture.adoc

```diff
@@ -7,19 +7,23 @@ order: 4
+SOTA Device Registry is a required component. While the original specification for SOTA called for clients to always be vehicles uniquely identified by VINs, there are automotive applications that don't necessarily want to use VINs as mandatory primary keys. The device registry keeps track of the client devices, and assigns each registered device a UUID, with VIN being an optional secondary identifier. Core creates update campaigns to vehicles/devices identified by their UUIDs, and so to create a campaign from a list of VINs (or other secondary identifiers) and corresponding software packages, it is necessary to query the device registry to resolve the UUID from the VIN and namespace.
 
-This project includes a fully functional and feature-complete resolver component, but the reason the resolver is architecturally separate from the core is to allow independent implementations by individual OEMs suitable to their own needs.
+SOTA Resolver is an optional component. Its primary purpose is, when given a packageID, to return a list of the devices that need that package. This allows the building of update campaigns. To accomplish that task, Resolver keeps an inventory of software packages and device characteristics (like, for example, hardware components/ECUs available or installed software versions). It also implements a filter engine to select device by characteristic, and permits the association of filters with packages. Thus, when Resolver is asked to resolve a packageID, it checks which filters are associated with the package, runs the filters against the device inventory, and returns a list of matching devices so that Core can create a properly-targeted update.
 
-The system architecture is illustrated here:
+Finally, SOTA contains a bare-bones admin interface. This web-based HMI exposes all of the functionality of the APIs of each component, and can be studied as an example of how to implement a more full-featured interface, or how to interact directly with the APIs. It also forwards API calls with valid session cookies on to the other components, and thus can be used as an entrypoint to the system while leaving the other microservices firewalled.
+
+In summary, the system looks a bit like this:
 
 image:../images/System-Architecture-Diagram.svg[System Architecture Diagram]
 
-The data model for the system is illustrated link:../images/Data-Model.svg[here].
+RVI clients would interact directly with Core's RVI node (note: needs to be deployed separately), while HTTPS clients would interact with the web server, which forwards their calls on to core/resolver/device_registry. Other authentication architectures are possible; ATS's commercial OTA offering uses SOTA as a base while implementing a more advanced security and authentication/authorization regime, with each API call getting authenticated directly by its final target. Other auth types, however, are out of scope of the open-source documentation.
```

Comment:
Isn't that only true with an OAuth2/Auth Plus setup? I'm just imagining, for example, Acer's deployment: since there's no authentication of API calls on core, presumably they'd want to set it up so that communication would have to be forwarded through web, and core wouldn't be world-accessible.